src/js/containers/Submission.jsx

```diff
@@ -35,51 +70,22 @@ class SubmissionContainer extends Component {
 
   render() {
     if(!this.props.user) return null
-    if(!this.props.status) return null
     if(!this.props.location) return null
 
-    if(this.props.status.code === null){
+    if(!this.props.isFetching &&
+      (!this.props.status || this.props.status.code === 0)){
       this.props.dispatch(fetchSubmission())
-      return null
     }
 
     const status = this.props.status
-    const code = status.code
+    const code = status && status.code
```

Comment:
These lines, ~80-84, might be cleaner as `const { status, params, user } = this.props`.